docs/recipes/precompiling-with-webpack.md

```diff
@@ -29,3 +37,226 @@ module.exports = {
 The important bits are `target: 'node'`, which ignores Node.js-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external Node.js modules which it may choke on.
 
 You can now run `$ ava _build/tests.js` to run the tests contained in this output.
+
+### Multiple test files
+Things are a little more complicated with multiple test files. We recommend [using babel-register](babelrc.md) until the performance penalty becomes too great.
+
+The possible approaches are:
+
+- [Refer precompiled source in tests](#refer-precompiled-source-in-tests)
+- [Single entry file](#single-entry-file)
+- [Multiple entry files](#multiple-entry-files)
+- [Multiple entry files with externalized source files](#multiple-entry-files-with-externalized-source-files)
+
+#### Refer precompiled source in tests
+Source files can be compiled with source maps(for coverage) to `_src` folder and referenced in tests. While this is less than elegant, it performs well and the worklow can be optimized with [webpack's watch mode](https://webpack.js.org/configuration/watch/).
+
+```js
+// Before
+import fresh from '../src';
+// After
+import fresh from '../_src';
+```
+
+#### Single entry file
+To pre-compile folder with multiple test files, we can use globbing(see [stackoverflow answer](http://stackoverflow.com/questions/32874025/how-to-add-wildcard-mapping-in-entry-of-webpack/34545812#34545812)). Although this performs quite possibly the best, it comes at a cost. AVA uses the filename and path for test names in verbose mode as well as test failure reports. This can make the error report harder to understand. Also, it can matter if tests modify globals, Node.js built-ins. You wouldn't want that to leak across tests.
```

Comment:
How about putting it like this?

> Multiple test files can be compiled into a single file. This may have the best performance but that does come at a cost. All tests will be in the same file, which can make it harder to know which test has failed, since AVA can't show the file name the test was originally in. You'll also lose [process isolation](https://github.com/avajs/ava#process-isolation).